packages/dd-trace/test/llmobs/sdk/index.spec.js

```diff
@@ -435,6 +438,180 @@ describe('sdk', () => {
+            setTimeout(() => {
+              cb(new Error('error'), 'output')
+            }, 1000)
+          }
+
+          const wrappedMyWorkflow = llmobs.wrap({ kind: 'workflow' }, myWorkflow)
+          wrappedMyWorkflow('input', (err, res) => {
+            expect(err).to.exist
+            expect(res).to.equal('output')
+          })
+
+          clock.tick(1000)
+
+          expect(LLMObsTagger.tagMap.get(span)).to.deep.equal({
+            '_ml_obs.meta.span.kind': 'workflow',
+            '_ml_obs.meta.ml_app': 'mlApp',
+            '_ml_obs.llmobs_parent_id': 'undefined',
+            '_ml_obs.meta.input.value': JSON.stringify({ input: 'input' }),
+            '_ml_obs.meta.output.value': 'output'
+          })
+        })
+
+        it('auto-annotates the first argument of the callback as the output if it is not an error', () => {
+          let span
+          function myWorkflow (input, cb) {
+            span = llmobs._active()
+            setTimeout(() => {
+              cb('output', 'ignore')
+            }, 1000)
+          }
+
+          const wrappedMyWorkflow = llmobs.wrap({ kind: 'workflow' }, myWorkflow)
+          wrappedMyWorkflow('input', (res, irrelevant) => {
+            expect(res).to.equal('output')
+            expect(irrelevant).to.equal('ignore')
+          })
+
+          clock.tick(1000)
+
+          expect(LLMObsTagger.tagMap.get(span)).to.deep.equal({
+            '_ml_obs.meta.span.kind': 'workflow',
+            '_ml_obs.meta.ml_app': 'mlApp',
+            '_ml_obs.llmobs_parent_id': 'undefined',
+            '_ml_obs.meta.input.value': JSON.stringify({ input: 'input' }),
+            '_ml_obs.meta.output.value': 'output'
+          })
+        })
+
+        it('maintains context consistent with the tracer', () => {
+          let llmSpan, workflowSpan, taskSpan
+
+          function myLlm (input, cb) {
+            llmSpan = llmobs._active()
+            setTimeout(() => {
+              cb(null, 'output')
+            }, 1000)
+          }
+          const myWrappedLlm = llmobs.wrap({ kind: 'llm' }, myLlm)
+
+          llmobs.trace({ kind: 'workflow', name: 'myWorkflow' }, _workflow => {
+            workflowSpan = _workflow
+            tracer.trace('apmOperation', () => {
+              myWrappedLlm('input', (err, res) => {
+                expect(err).to.not.exist
+                expect(res).to.equal('output')
+                llmobs.trace({ kind: 'task', name: 'afterLlmTask' }, _task => {
+                  taskSpan = _task
+
+                  const llmParentId = LLMObsTagger.tagMap.get(llmSpan)['_ml_obs.llmobs_parent_id']
+                  expect(llmParentId).to.equal(workflowSpan.context().toSpanId())
+
+                  const taskParentId = LLMObsTagger.tagMap.get(taskSpan)['_ml_obs.llmobs_parent_id']
+                  expect(taskParentId).to.equal(workflowSpan.context().toSpanId())
+                })
+              })
+            })
+          })
+        })
+
```

Comment:
:+1: excellent job with these tests @sabrenner! They were concise and easy to understand.